### PR TITLE
Fix documentation for deprecated 'reports' object (Issue #5908)

### LINKED ISSUE #5908: Documentation suggests using deprecated method
https://detekt.dev/docs/introduction/reporting/#kotlin-dsl

The documentation for reporting gives an example using `reports.xml.required.set(true)`, however the `reports` object is marked as deprecated.

### PR DESCRIPTION
**Pull request by BeBAKE**
I made a completely new branch and removed whitespaced and fixed the deprecated reports object.

Fixes #5908

### CODE REVIEW

File: website/versioned_docs/version-1.22.0/introduction/reporting.md [hunk 1]
```diff
@@ -93,8 +93,10 @@ tasks.register("reportMerge", io.gitlab.arturbosch.detekt.report.ReportMergeTask
 
 subprojects {
   detekt {
-    reports.xml.required.set(true)
-    // reports.sarif.required.set(true)
+    reports {
+      xml.required.set(true)
+      // sarif.required.set(true)
+    }
   }
```
Comment (cortinico) [inline website/versioned_docs/version-1.22.0/introduction/reporting.md:100]: The docs are still wrong. The problem is that the whole:

```diff
detekt {
-   reports {
-        ...
-   }
}
```

is deprecated. Reports should be configured using this approach:
https://detekt.dev/docs/gettingstarted/gradle#reports

**Review by cortinico — APPROVED** [review]
LGTM 👍

**Comment by BeBAKE** [comment]
should i write a test case for the [detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt](https://app.codecov.io/gh/detekt/detekt/pull/5924/blob/detekt-formatting/src/main/kotlin/io/gitlab/arturbosch/detekt/formatting/FormattingRule.kt)
line 97

**Comment by BraisGabin** [comment]
That's not necessary. Thanks to improve the documentation

**Pull request merged**